zkevm-circuits/src/state_circuit/constraint_builder.rs

```diff
@@ -286,18 +361,35 @@ impl<F: Field> Queries<F> {
         self.field_tag.clone()
     }
 
+    fn tag_change(&self) -> Expression<F> {
+        self.tag() - self.prev_tag.clone()
+    }
+
     fn value(&self) -> Expression<F> {
         self.value.clone()
     }
 
-    fn tag_matches(&self, tag: RwTableTag) -> Expression<F> {
+    fn tag_matches(&self, tag: RwTableTag, is_prev: bool) -> Expression<F> {
```

Comment:
less easier to read code ....

better to change method to 

```
fn tag_matches(&self, tag: RwTableTag, tag_expr: Expression<F>) -> Expression<F> {
     generate_lagrange_base_polynomial(
            tag_expr,
            tag as usize,
            RwTableTag::iter().map(|x| x as usize),
        )
```

Reply:
as import is_tag_unchanged no need to change this method now,  remain it unchanged now .